kii_thing_if.c

```diff
@@ -37,6 +37,60 @@
 
 #define THING_IF_INFO "sn=tic;sv=0.9.6"
 
+typedef enum successor_t {
+    NOT_SUCCESSOR,
+    IS_SUCCESSOR
+} successor_t;
+
+static int prv_append_key_value_string(
+        kii_t* kii,
+        const char* key,
+        const char* value,
+        successor_t successor)
+{
+    if (key == NULL) {
+        M_KII_LOG(kii->kii_core.logger_cb("key not specified.\n"));
+        return -1;
+    }
+    if (value == NULL) {
+        M_KII_LOG(
+            kii->kii_core.logger_cb("value not specified for key: %s.\n", key));
+        return -1;
+    }
+
+    if (successor == IS_SUCCESSOR) {
+        if (kii_api_call_append_body(kii, ",", CONST_STRLEN(",") != 0)) {
+            M_KII_LOG(kii->kii_core.logger_cb(
+                "request size overflowed: (%s, %s).\n", key, value));
+            return -1;
+        }
+    }
+    if (kii_api_call_append_body(kii, "\"", CONST_STRLEN("\"")) != 0 ||
+            kii_api_call_append_body(kii, key, strlen(key)) != 0 ||
+            kii_api_call_append_body(
+                kii, "\":\"", CONST_STRLEN("\":\"")) != 0 ||
+            kii_api_call_append_body(kii, value, strlen(value)) != 0 ||
+            kii_api_call_append_body(kii, "\"", CONST_STRLEN("\"")) != 0) {
+        M_KII_LOG(kii->kii_core.logger_cb(
+            "request size overflowed: (%s, %s).\n", key, value));
+        return -1;
+    }
+    return 0;
+}
+
+static int prv_append_key_value_string_optional(
+        kii_t* kii,
+        const char* key,
+        const char* value,
+        successor_t successor)
+{
+    if (key == NULL || value == NULL) {
+        return 0;
+    }
+    return prv_append_key_value_string(kii, key, value, successor);
+}
+
+
```

Comment:
@yongpingchen 

Using 3rd party library is one candidate. We might write code easier than this  but I think binary size and required stack of this SDK size become bigger than this. thing-if ThingSDK is for things. They might only have limited memory. 

This is trade-off.  We should choose binary size and required stack in this case because things only have limited memory.

What do you think?